lambda/simplif.ml

```diff
@@ -119,69 +119,69 @@ let simplify_exits lam =
         Hashtbl.add exits i r
   in
 
-  let rec count = function
+  let rec count depth = function
   | (Lvar _| Lconst _) -> ()
-  | Lapply ap -> count ap.ap_func; List.iter count ap.ap_args
-  | Lfunction {body} -> count body
+  | Lapply ap -> count depth ap.ap_func; List.iter (count depth) ap.ap_args
+  | Lfunction {body} -> count depth body
   | Llet(_str, _kind, _v, l1, l2) ->
-      count l2; count l1
+      count depth l2; count depth l1
   | Lletrec(bindings, body) ->
-      List.iter (fun (_v, l) -> count l) bindings;
-      count body
-  | Lprim(_p, ll, _) -> List.iter count ll
+      List.iter (fun (_v, l) -> count depth l) bindings;
+      count depth body
+  | Lprim(_p, ll, _) -> List.iter (count depth) ll
   | Lswitch(l, sw, _loc) ->
-      count_default sw ;
-      count l;
-      List.iter (fun (_, l) -> count l) sw.sw_consts;
-      List.iter (fun (_, l) -> count l) sw.sw_blocks
+      count_default depth sw ;
+      count depth l;
+      List.iter (fun (_, l) -> count depth l) sw.sw_consts;
+      List.iter (fun (_, l) -> count depth l) sw.sw_blocks
   | Lstringswitch(l, sw, d, _) ->
-      count l;
-      List.iter (fun (_, l) -> count l) sw;
+      count depth l;
+      List.iter (fun (_, l) -> count depth l) sw;
       begin match  d with
       | None -> ()
       | Some d -> match sw with
-        | []|[_] -> count d
-        | _ -> count d; count d (* default will get replicated *)
+        | []|[_] -> count depth d
+        | _ -> count depth d; count depth d (* default will get replicated *)
       end
-  | Lstaticraise (i,ls) -> incr_exit i 1 !try_depth; List.iter count ls
+  | Lstaticraise (i,ls) -> incr_exit i 1 try_depth; List.iter (count depth) ls
   | Lstaticcatch (l1,(i,[]),Lstaticraise (j,[])) ->
       (* i will be replaced by j in l1, so each occurrence of i in l1
          increases j's ref count *)
-      count l1 ;
+      count depth l1 ;
       let ic = get_exit i in
-      incr_exit j ic.count (max !try_depth ic.max_depth)
+      incr_exit j ic.count (max try_depth ic.max_depth)
   | Lstaticcatch(l1, (i,_), l2) ->
-      count l1;
+      count depth l1;
       (* If l1 does not contain (exit i),
          l2 will be removed, so don't count its exits *)
       if (get_exit i).count > 0 then
-        count l2
-  | Ltrywith(l1, _v, l2) -> incr try_depth; count l1; decr try_depth; count l2
-  | Lifthenelse(l1, l2, l3) -> count l1; count l2; count l3
-  | Lsequence(l1, l2) -> count l1; count l2
-  | Lwhile(l1, l2) -> count l1; count l2
-  | Lfor(_, l1, l2, _dir, l3) -> count l1; count l2; count l3
-  | Lassign(_v, l) -> count l
-  | Lsend(_k, m, o, ll, _) -> List.iter count (m::o::ll)
-  | Levent(l, _) -> count l
-  | Lifused(_v, l) -> count l
-
-  and count_default sw = match sw.sw_failaction with
+        count depth l2
+  | Ltrywith(l1, _v, l2) -> ignore (try_depth+1); count depth l1; ignore (try_depth-1); count depth l2
```

Comment:
You're still using `try_depth` instead of the `depth` parameter here (and in a few other places).
Using `ignore` is also a bad sign.